src/test/java/io/supertokens/test/accountlinking/GetUserByAccountInfoTest.java

```diff
@@ -0,0 +1,366 @@
+        Passwordless.CreateCodeResponse code = Passwordless.createCode(main, email, null,
+                null, "123456");
+        return Passwordless.consumeCode(main, code.deviceId, code.deviceIdHash,
+                code.userInputCode, null).user;
+    }
+
+    AuthRecipeUserInfo createPasswordlessUserWithPhone(Main main, String phone)
+            throws DuplicateLinkCodeHashException, StorageQueryException, NoSuchAlgorithmException, IOException,
+            RestartFlowException, InvalidKeyException, Base64EncodingException, DeviceIdHashMismatchException,
+            StorageTransactionLogicException, IncorrectUserInputCodeException, ExpiredUserInputCodeException {
+        Passwordless.CreateCodeResponse code = Passwordless.createCode(main, null, phone,
+                null, "123456");
+        return Passwordless.consumeCode(main, code.deviceId, code.deviceIdHash,
+                code.userInputCode, null).user;
+    }
+
+    @Test
+    public void testListUsersByAccountInfoForUnlinkedAccounts() throws Exception {
+        String[] args = {"../"};
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args, false);
+        FeatureFlagTestContent.getInstance(process.getProcess())
+                .setKeyValue(FeatureFlagTestContent.ENABLED_FEATURES, new EE_FEATURES[]{
+                        EE_FEATURES.ACCOUNT_LINKING, EE_FEATURES.MULTI_TENANCY});
+        process.startProcess();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        AuthRecipeUserInfo user1 = createEmailPasswordUser(process.getProcess(), "test1@example.com", "password1");
+        AuthRecipeUserInfo user2 = createThirdPartyUser(process.getProcess(), "google", "userid1", "test2@example.com");
+        AuthRecipeUserInfo user3 = createPasswordlessUserWithEmail(process.getProcess(), "test3@example.com");
+        AuthRecipeUserInfo user4 = createPasswordlessUserWithPhone(process.getProcess(), "+919876543210");
+
+        TenantIdentifierWithStorage tenantIdentifierWithStorage = TenantIdentifier.BASE_TENANT.withStorage(StorageLayer.getBaseStorage(process.getProcess()));
+
+        // test for result
+        assertEquals(user1, AuthRecipe.getUsersByAccountInfo(tenantIdentifierWithStorage, false, "test1@example.com", null, null, null)[0]);
+        assertEquals(user2, AuthRecipe.getUsersByAccountInfo(tenantIdentifierWithStorage, false, null, null, "google", "userid1")[0]);
+        assertEquals(user2, AuthRecipe.getUsersByAccountInfo(tenantIdentifierWithStorage, false, "test2@example.com", null, "google", "userid1")[0]);
+        assertEquals(user3, AuthRecipe.getUsersByAccountInfo(tenantIdentifierWithStorage, false, "test3@example.com", null, null, null)[0]);
+        assertEquals(user4, AuthRecipe.getUsersByAccountInfo(tenantIdentifierWithStorage, false, null, "+919876543210", null, null)[0]);
```

Comment:
you also should test the structure of the output for at least one of these.

Reply:
Done